applications/DEM_application/custom_conditions/analytic_RigidFace.cpp

```diff
@@ -0,0 +1,129 @@
+// Author: Guillermo Casas (gcasas@cimne.upc.edu)
+
+// Project includes
+#include "custom_conditions/analytic_RigidFace.h"
+#include "../custom_elements/spheric_particle.h"
+
+namespace Kratos {
+
+// Constructor
+
+AnalyticRigidFace3D::AnalyticRigidFace3D() : mNumberOfCrossingSpheres(0){}
+
+// Constructor
+
+AnalyticRigidFace3D::AnalyticRigidFace3D(IndexType NewId, GeometryType::Pointer pGeometry) : RigidFace3D(NewId, pGeometry), mNumberOfCrossingSpheres(0){}
+
+// Constructor
+
+AnalyticRigidFace3D::AnalyticRigidFace3D(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties)
+           : RigidFace3D(NewId, pGeometry, pProperties), mNumberOfCrossingSpheres(0){}
+
+
+//***********************************************************************************
+//***********************************************************************************
+
+Condition::Pointer AnalyticRigidFace3D::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    return Condition::Pointer(new AnalyticRigidFace3D(NewId, GetGeometry().Create(ThisNodes), pProperties));
+}
+
+// Destructor
+
+AnalyticRigidFace3D::~AnalyticRigidFace3D(){}
+
+
+void AnalyticRigidFace3D::ComputeConditionRelativeData(int rigid_neighbour_index,
```

Comment:
No, this is a left over from a previous design that required overriding. I will remove it in the next pull request, since this one will be cancelled